src/lib.rs

```diff
@@ -0,0 +1,4 @@
+pub trait Residency {
+    fn get_exchange_rate(&self, transaction_date: &str) -> Result<(String, f32), String>;
```

Comment:
You could consider changing return type from Result to simpler one until you add returning of errors just for simplicity.